src/history/hs.c

```diff
@@ -564,11 +565,14 @@ __wt_hs_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MUL
     int nentries;
     bool squashed;
 
-    session = CUR2S(cursor);
+    btree = S2BT(session);
+    cursor = session->hs_cursor;
     prev_upd = NULL;
     insert_cnt = 0;
     __wt_modify_vector_init(session, &modifies);
 
+    WT_ASSERT(session, cursor != NULL);
+
```

Comment:
I don't see much point in this assert, as we're going to drop core in about 10 instructions if cursor is NULL, right? I don't feel strongly, though.

Reply:
Removed